app/controllers/provider/admin/backend_apis_controller.rb

```diff
@@ -37,7 +37,7 @@ def update
   end
 
   def destroy
-    if @backend_api.services.empty? && @backend_api.mark_as_deleted
+    if @backend_api.mark_as_deleted
       redirect_to provider_admin_dashboard_path, notice: 'Backend will be deleted shortly.'
     else
       flash[:error] = 'Backend could not be deleted'
```

Comment:
```suggestion
      flash[:error] = 'Backend could not be deleted because it's used in a Product'
```

Reply:
Imo that's not a good idea. Then we should show the flash containing he errors in the model, because that is not the only possible reason why it couldn't be deleted.

Reply:
I mean like: `flash[:error] = @backend_api.errors.full_messages.to_sentence`